curriculum/challenges/english/25-front-end-development/review-bash-scripting/6724e417419c2f211bb41bfc.md

```diff
@@ -7,7 +7,708 @@ dashedName: review-bash-scripting
+
+  ```bash
+  NAME=John
+  MESSAGE="Hello World"
+  COUNT=5
+  TEXT="The next number is, "
+  ```
+
+- **Variable usage**: Access variable values by placing `$` in front of the variable name.
+
+  ```bash
+  echo $NAME
+  echo "The message is: $MESSAGE"
+  ```
+
+- **Variable interpolation**: Use `$variable_name` to access the value of a variable within strings and commands.
+
+```bash
+TEXT="The next number is, "
+NUMBER=42
+echo $TEXT B:$NUMBER
+echo $TEXT I:$NUMBER
+
+echo "Pulling $server"
+rsync --archive --verbose $server:/etc/nginx/conf.d/server.conf configs/$server.conf
+```
+
+- **Variable scope**: Shell scripts run from top to bottom, so variables can only be used below where they are created.
+
+  ```bash
+  NAME="Alice"
+  echo $NAME
+  ```
+
+- **User input**: Use `read` to accept input from users and store it in a variable.
+
+  ```bash
+  read USERNAME
+  echo "Hello $USERNAME"
+  ```
+
+- **Comments**: Add comments to your scripts using `#` followed by your comment text.
+  - Single-line comments start with `#` and continue to the end of the line
+  - Comments are ignored by the shell and don't affect script execution
+
+  ```bash
+  # This is a single-line comment
+  NAME="John"  # Comment at end of line
+  ```
+
+- **Multi-line comments**: Comment out blocks of code using colon and quotes.
+
+  ```bash
+  : '
+  This is a multi-line comment
+  Everything between the quotes is ignored
+  Useful for debugging or documentation
+  '
+  ```
+
+- **Built-in commands and help**: 
+  - Use `help` to see a list of built-in bash commands
+  - Use `help <command>` to get information about specific built-in commands
+  - Some commands (like `if`) are built-ins and don't have man pages
+  - Built-in commands are executed directly by the shell rather than as external programs
+  - Use `help function` to see information about creating functions
+
+  ```bash
+  help
+  help if
+  help function
+  ```
+
+- **Finding command locations**: Use `which` to locate where executables are installed.
+  - Shows the full path to executable files
+  - Useful for finding interpreter locations (like bash)
+  - Helps verify which version of a command will be executed
+
+  ```bash
+  which bash
+  which python
+  which ls
+  ```
+
+- **Manual pages**: Use `man` to access detailed documentation for commands.
+  - Provides comprehensive information about command usage
+  - Shows all available options and examples
+  - Use arrow keys to navigate, 'q' to quit
+  - Not all commands have manual pages (built-ins use `help` instead)
+
+  ```bash
+  man echo
+  man ls
+  man bash
+  ```
+
+- **Help flags**: Many commands support `--help` for quick help information.
+  - Alternative to manual pages for quick reference
+  - Shows command syntax and common options
+  - Not all commands support this flag (some may show error)
+
+  ```bash
+  ls --help
+  chmod --help
+  echo --help
```

Comment:
`echo --help` simply prints `--help` to the screen:
<img width="515" height="68" alt="image" src="https://github.com/user-attachments/assets/071ba808-0810-41c0-a6ea-11c5977d6bfb" />

This should be replaced with a working example:

```suggestion
  mv --help
```